### PR TITLE
rewrite recipe; update to 3.2.1

### LINKED ISSUE #16: Update LightGBM to 3.2.1
https://github.com/microsoft/LightGBM/releases/tag/v3.2.1

py39 support was added in 3.2.0  https://github.com/microsoft/LightGBM/pull/3693

upstream workflows currently run against cuda 11.2.2

### PR DESCRIPTION
**Pull request by jayfurmanek**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any [tests](https://github.com/open-ce/open-ce/blob/main/tests/) to validate this change?  

## Description

Fixes #16 
recipe rewrite:
  - dropped "select" and "meta" recipes: consolidate meta package into single recipe
  - cuda variant supported via build string
  - added conda hash to build string

new package listing:
```
liblightgbm-3.2.1-h2fbd448_1_cpu_openmpi.tar.bz2
liblightgbm-3.2.1-h77b50d1_1_cuda10.2_openmpi.tar.bz2
lightgbm-3.2.1-py37h2224f93_1_cpu_openmpi.tar.bz2
lightgbm-3.2.1-py37h4e16e45_1_cuda10.2_openmpi.tar.bz2
lightgbm-proc-3.2.1-cpu.tar.bz2
lightgbm-proc-3.2.1-cuda.tar.bz2
```

It might be easiest to review from the source: github.com/jayfurmanek/LightGBM-feedstock

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,18 +16,40 @@ source:
...
-      string: {{ build_type }}_{{ mpi_type }}_{{ PKG_BUILDNUM }} #[build_type == 'cpu']
-      string: {{ build_type }}{{ cudatoolkit | replace(".*", "") }}_{{ mpi_type }}_{{ PKG_BUILDNUM }} #[build_type == 'cuda']
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ build_type }}_{{ mpi_type }} #[build_type == 'cpu']
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ build_type }}{{ cudatoolkit | replace(".*", "") }}_{{ mpi_type }} #[build_type == 'cuda']
 {% if build_type == 'cuda' %}
       script_env:
         - CUDA_HOME
 {% endif %}
+      run_exports:
+        - {{ pin_subpackage("liblightgbm", max_pin="x.x.x") }}
```
Comment (bnemanich) [inline recipe/meta.yaml]: Why is this being added?

Comment (jayfurmanek) [reply recipe/meta.yaml]: Yeah, that seems like it pins to itself.
We have a similar line in the arrow recipe which is mostly what this update is based on.
https://github.com/open-ce/pyarrow-feedstock/blob/main/recipe/meta.yaml#L46-L47

I'll try to remove that and see what happens.

Comment (jayfurmanek) [reply recipe/meta.yaml]: It seems to work just fine without this explicit run_export.
I'll remove it.

**Comment by cdeepali** [comment]
I got successful build on:
```
RH8 x86 baremetal with CUDA 11.0 
RH7 ppc container with CUDA 10.2 
```

**Review by bnemanich — APPROVED** [review]

**Pull request merged**